### PR TITLE
Added new attributes StoragePoolName and DataCenterName in volume's project API response.

### PR DESCRIPTION
**Pull request by chitranm**
After enabling the import of external volumes in Metal, it is possible that Volume is created on site A can be exported/attached to Host created on site B. And in this case volume will be seen in compute group where host is created.

Currently BMaaS UI uses **/available-resources** project API end point to fill StoragePoolName and DataCenterName. But **/available-resources** API always returns the data specific to the site. And right now for above described scenario, these fields are showing empty as volume is created on different site than the host. 

For more details please see the defect - DE24521.

### CODE REVIEW

**Review by dbozzato81 — COMMENTED** [review]
Can you please clarify what's the benefits/reason behind this change since model already contains : StoragePoolID and LocationID?

File: v1/api/swagger/components/schemas/Volume.yaml [hunk 1]
```diff
@@ -36,6 +38,10 @@ allOf:
         description: >-
           The storage pool ID matching an entry in the StoragePools array
           returned as part of the get /available-resources call
+      StoragePoolName:
+        type: string
+        description: >-
+          StoragePoolName is the name of storage pool from where the volume is allocated
```
Comment (anurag-shrivastav) [inline v1/api/swagger/components/schemas/Volume.yaml:44]:
```suggestion
          Name of storage pool from where the volume is allocated.
```

File: v1/api/swagger/components/schemas/Volume.yaml [hunk 2]
```diff
@@ -59,6 +65,10 @@ allOf:
           of those listed by the LocationInfo array returned as part of the get
           /available-resources call. Any volumes must be in the same location as
           their attached Host.
+      DataCenterName:
+        type: string
+        description: >-
+          DataCenterName is the name of the datacenter where the volume is created on the storage array
```
Comment (anurag-shrivastav) [inline v1/api/swagger/components/schemas/Volume.yaml:71]:
```suggestion
          Name of the data center where the volume is created on the storage array.
```

**Comment by chitranm** [comment]
> Can you please clarify what's the benefits/reason behind this change since model already contains : StoragePoolID and LocationID?

@dbozzato81, Since we don't have **StoragePoolName** and **DataCenterName** in project API response of volume, and BMaaS UI wants to show the name instead of IDs, BMaaS UI calls available-resources project API end point. And it returns the data specific to the site. 

After enabling the import of external volumes in Metal, it is possible that Volume is created on site A can be exported/attached to Host created on site B. And in this case volume will be seen in compute group where host is created in BMaaS UI. 

In the above case volume will be seen in compute group of host, but location of the volume will be set to other site and storage pool location will be set to other site. And due to this, it will show empty **StoragePoolName** and wrong pod name (Current behavior seen in Mediahuis in some of the volumes).

It can be fixed using **available-resources**  API if we return locations and storage pools irrespective of site but it will be hacky way of fixing it (will have to check if feature is enabled or not for the hoster). 

If you are still not convinced, we can have a call.

**Review by anurag-shrivastav — COMMENTED** [review]
LGTM. Will let Denis approve.

**Review by dbozzato81 — APPROVED** [review]

**Comment by karthik-radakrishnan** [comment]
Denis, Malay and I discussed this earlier today and we are good to go. Approving.

**Review by karthik-radakrishnan — APPROVED** [review]

**Pull request merged**